src/main/sensors/voltage.c

```diff
@@ -201,7 +206,11 @@ void voltageMeterADCRefresh(void)
 void voltageMeterADCRead(voltageSensorADC_e adcChannel, voltageMeter_t *voltageMeter)
 {
     voltageMeterADCState_t *state = &voltageMeterADCStates[adcChannel];
-
+    
+    if (millis() - voltageMeter->prevDisplayTime > PREV_DISPLAY_FILTERED_TIME_DIFF) {
```

Comment:
This is a bit problematic - value is stored at 1s intervals (samples at 50Hz), but comparison is done at batteryUpdatePresence (20Hz) rate. So distance between compared samples will be dependent on synchronization of both tasks (and then increased over 1s interval).

Either compare only once per second(per PREV_DISPLAY_FILTERED_TIME_DIFF) so that compared samples are PREV_DISPLAY_FILTERED_TIME_DIFF apart, or compare each filtered sample (roughly equivalent to current worst case) or use some better statistics (sum sqr() of differences over PREV_DISPLAY_FILTERED_TIME_DIFF interval?)